logstash-core/lib/logstash/instrument/metric.rb

```diff
@@ -4,36 +4,48 @@
 require "concurrent"
 
 module LogStash module Instrument
-  class MetricNoKeyProvided < Exception; end
-  class MetricNoBlockProvided < Exception; end
+  class MetricException < Exception; end
+  class MetricNoKeyProvided < MetricException; end
+  class MetricNoBlockProvided < MetricException; end
+  class MetricNoNamespaceProvided < MetricException; end
 
   class Metric
-    attr_reader :collector, :base_key
+    attr_reader :collector, :namespace_information
 
-    def initialize(collector, base_key = nil)
+    public
+    def initialize(collector, namespace = nil)
       @collector = collector
-      @base_key = base_key
+      @namespace_information = Array(namespace).map(&:to_sym)
```

Comment:
do we aim to introduce some kind of separator between the different namespace components? this will be useful when performing operations in the snapshot to organize the data.